nvflare/private/fed/app/fl_app_validator.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import Dict, Tuple, Optional, List
+
+from nvflare.apis.app_validation import AppValidator, AppValidationKey
+
+
+class DefaultAppValidator(AppValidator):
+    def validate(self, app_folder: str) -> Tuple[str, Dict]:
+        result = dict()
+        app_root = os.path.abspath(app_folder)
+        if not os.path.exists(os.path.join(app_root, "config")):
+            return "Missing config folder inside app folder.", {}
+        if not os.path.exists(os.path.join(app_root, "config", "config_fed_server.json")):
```

Comment:
In theory, if the app is deployed to Server, then only config_fed_server.json is needed; if the app is for Client, then only config_fed_client.json is needed. However this is hard to do at this time since we don't know what the app is to be used for. In NVF2.1, we'll know this and the validation will be done for the whole "experiment" that contains all apps and sufficient info.

So let's just keep this in mind.